### PR TITLE
[ShapeFunc] Handle weights in shape func

### PR DESCRIPTION
**Pull request by anijain2305**
When compiling a dynamic model with TRT, the primitive function has constants already bound in the graph. TVM codegen typically lifts those into variables in ANF pass, so we never see this problem in TVM codegen. 

@icemelon9 @zhiics @codeislife99 @rohanmukh @trevor-m

### CODE REVIEW

**Review by zhiics — APPROVED** [review]
LGTM. @icemelon9 PTAL

File: src/relay/backend/compile_engine.cc [hunk 1]
```diff
@@ -420,38 +420,59 @@ class MakeShapeFunc : public backend::MemoizedExprTranslator<Array<te::Tensor>>
...
-          "data_const", topi::kBroadcast);
-      scalars_.push_back(value);
-      return {value};
-    } else {
-      auto value = tvm::te::compute(
-          {}, [&](const Array<tvm::tir::Var>&) { return tir::make_const(DataType::Int(64), 0); },
           "shape_const", topi::kBroadcast);
       scalars_.push_back(value);
       return {value};
+    } else {
```
Comment (icemelon9) [inline src/relay/backend/compile_engine.cc:444]: No need for else since you already return.

**Review by icemelon9 — APPROVED** [review]
LGTM. just a minor comment.

**Comment by anijain2305** [comment]
Thanks @zhiics @icemelon9 This is merged.

**Pull request merged**